rfcs/chromium-connector.md

```diff
@@ -0,0 +1,213 @@
+
+```json
+{
+  "connector": {
+    "name": "chromium",
+    "options": {
+     // Each connector decides its options
+    }
+  }
+}
+```
+
+This will most likely be a **breaking** change in the connectors.
+
+### Allow content as a target
+
+We recently created #2149 so connectors can tell if they accept content directly or not. To support this there's [`page.setContent(html[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagesetcontenthtml-options).
+
+There should not be a high usage for this feature but it seems easy enough to be supported (even if it is not on the first version).
+
+
+## `connector-chromium`-`Puppeteer` API mapping
+
+This is a (non-exhaustive) list of all the APIs to use:
+
+| Action | API |
+| ------ | --- |
+| Start  | [`puppeteer.launch([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerlaunchoptions) <br> [`puppeteer.connect(options)`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-puppeteerconnectoptions)|
+| JS evaluation         | [`page.evaluate(pageFunction[, ...args])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pageevaluatepagefunction-args) |
+| Request notification  | [`page.on('request')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-request) <br> [`page.on('requestfailed')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-requestfailed) <br> [`page.on('requestfinished')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-requestfinished) |
+| Page errors | [`page.on('error)`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-error) <br> [`page.on('pageerror)`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-pageerror) (for uncaught exceptions) |
+| Response notification | [`page.on('response')`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-event-response) |
+| Close browser         | [`browser.close()`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-browserclose) |
+| Navigate              | [`browser.newPage()`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-browsernewpage) <br> [`page.goto(url[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagegotourl-options) <br> [`page.waitForNavigation([options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagewaitfornavigationoptions) |
+| Set page content | [`page.setContent(html[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagesetcontenthtml-options) |
+| Input | [`page.type(selector, text[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pagetypeselector-text-options) <br> [`page.click(selector[, options])`](https://pptr.dev/#?product=Puppeteer&version=v1.14.0&show=api-pageclickselector-options) |
+
+
+## Other comments/questions
+
+**How to run multiple pages in the same browser from different Puppeteer instances?**
```

Comment:
Any examples of when we would want/need to do this?

Reply:
When we are running the tests basically 😊